src/command.coffee

```diff
@@ -185,7 +186,7 @@ compileScript = (file, input, base = null) ->
     else
       compiled = CoffeeScript.compile t.input, t.options
       t.output = compiled
-      if o.map
+      if o.map and not o['inline-map']
```

Comment:
@XVincentX I am no longer using coffee-script for any new projects, just maintaining old ones.

This PR does fix the issue, but the core contributors haven't been able to reach a decision to merge.